cloudinit/config/cc_growpart.py

```diff
@@ -333,6 +456,49 @@ def resize_devices(resizer, devices):
             )
             continue
 
+        underlying_blockdev = get_mapped_device(blockdev)
+        if underlying_blockdev:
+            try:
+                # We need to resize the underlying partition first
+                partition = get_underlying_partition(blockdev)
+                if is_encrypted(underlying_blockdev, partition):
+                    if partition not in [x[0] for x in info]:
+                        # We shouldn't attempt to resize this mapped partition
+                        # until the underlying partition is resized, so re-add
+                        # our device to the beginning of the list we're
+                        # iterating over, then add our underlying partition
+                        # so it can get processed first
+                        devices.insert(0, devent)
+                        devices.insert(0, partition)
```

Comment:
Why are we re-inserting the partition into devices list? Isn't out only desire here to make sure we still process the base devent after our continue below?

Can avoid this second insert?
```suggestion
```

Reply:
Answered in side-channel. I had overlooked the immediate continue just below this line. So, this first iteration didn't resize anything when determining the mapped device and partition. So enqueuing the partition to resize is needed to actual make those changes to the partiiton. The the subsquent resize of the blockdevice can take place on 2nd iteration of the loop.